src/main/java/com/zoo/species/Elephant.java

```diff
@@ -1,22 +1,53 @@
 package com.zoo.species;
 
 import com.zoo.animals.Mammal;
+import com.zoo.interfaces.IPlayable;
 
-public class Elephant extends Mammal{
-
-    // Constructor
-    public Elephant(String name, int age, double weight, int energyLevel, String furColor) {
-        super(name, age, weight, energyLevel, furColor);
-        this.furColor = furColor;
+public class Elephant extends Mammal implements IPlayable {
+    //Constructor
+    public Elephant(String name, int energyLevel, String furColor) {
+        super(name, energyLevel, furColor);
     }
-    //Method override
+
     @Override
-    public void makeSound () {
+    public void makeSound() {
         System.out.println(name + " says: Truba-bа-bа! ");
     }
-// Adding a new method
-public void spraySelf() {
-        System.out.println(name + " is spraying intself with water from its trunk.  ");
+
+    @Override
+    public void eat() {
+        setEnergyLevel(Math.min(getEnergyLevel() + 20, 100));
+        System.out.println(name + " is eating grass and leaves. Energy is now " + getEnergyLevel() + ".");
+    }
+
+    @Override
+    public void sleep() {
+        setEnergyLevel(Math.min(getEnergyLevel() + 30, 100));
```

Comment:
You see you have to repeat this construction so many times?

The `Math.min` thing is not bad overall, the bad thing here is you're "forced" to do this validation to keep the energy level in bounds [0, 100]

My proposal was (and you did it) -> to move all responsibility for managing the bounds to a single place, a family of "Energy control" methods which includes setter of energyLevel itself plus 2 additional methods `increaseEnergyBy(int amount)` and `decreaseEnergyBy(int amount)`

By doing this, in the rest of the code you would just hold Aperol in one hand, and call appropriate methods when needed to increase/decrease energy without worrying about the bounds :)